add test class for keto meal.

Hossam-Haroon opened the pull request:
add all the possible test cases for keto meal useCase

Hossam-Haroon committed:
make a test class for keto meal.
add all the possible test cases.

Mohamed-Elhanafy commented on build.gradle:
```diff
@@ -14,6 +15,11 @@ dependencies {
     testImplementation 'org.jetbrains.kotlin:kotlin-test'
     implementation 'io.insert-koin:koin-core:4.0.3'
     implementation 'org.jetbrains.kotlinx:kotlinx-datetime:0.6.2'
+    //testImplementation 'org.junit.jupiter:junit-jupiter:5.8.1'
```
you need to resolve the conflicts and get the dependencies from the remote

Hossam-Haroon replied:
made a note on slack

Mohamed-Elhanafy commented on build.gradle:
```diff
@@ -1,5 +1,6 @@
 plugins {
     id 'org.jetbrains.kotlin.jvm' version '2.1.10'
+    //id 'org.jetbrains.kotlin.jvm' version '1.9.10'
```
I think we can remove this

Mohamed-Elhanafy commented on src/test/kotlin/GetKetoDietMealUseCaseTest.kt:
```diff
@@ -0,0 +1,86 @@
+        val result = getKetoDietMealUseCase.getSuggestedKetoMeal(setOf())
+
+        // Then
+        assertThat(result.isFailure).isTrue()
```
I think it's better to specify the exception

Hossam-Haroon replied:
how ?

ahmednasserzaza replied:
@Mohamed-Elhanafy @Hossam-Haroon 
It's okay for this test case to test if isFailure invoked or not, also you should create another test case to test if a specific exception is throwed or not

Mohamed-Elhanafy replied:
@Hossam-Haroon  
`assertThat(result.exceptionOrNull()).isInstanceOf(MealException.NoMealsFoundException::class.java)`
this how you test if your exception is instance of our custom exception 

@ahmednasserzaza In this case is it better to put two assertions in the testcase ? I know that Bareq mentioned it in yesterday's review, but in chapter 9 of the Clean Code was talking about the one assertion per test, and he also mentioned that it's okay if we use more than one assertion if it is in the same context.

ahmednasserzaza commented on src/test/kotlin/CreateKetoMealHelper.kt:
```diff
@@ -0,0 +1,27 @@
+import kotlinx.datetime.LocalDate
+
+
+fun createMeal(
```
Wrong naming for file 
it should be bettter if we made it KetoMealHelper,
I prefer making this function into the same test class in a companion object because its specific for KetoMels only

Hossam-Haroon replied:
I started it with a verb t okeep it clean based on the book also I remember bariq called it like this in his videos createCity, and I make a meal in general keto or not. so what do you think ?

ahmednasserzaza reviewed (approved):
Well done
Please check the comments

Hossam-Haroon committed:
change the name of the function that creates fake meals.
transfer the function to the test class inside companion object
add a test function to test if the function throws an exception

Hossam-Haroon committed:
change the name of the function that creates fake meals.
transfer the function to the test class inside companion object
add a test function to test if the function throws an exception

Hossam-Haroon committed:
change the name of the function that creates fake meals.
transfer the function to the test class inside companion object
add a test function to test if the function throws an exception
